### DISCUSSION TITLE
how to import data to self-hosted umami

### DISCUSSION THREAD

**Discussion by chinesehuazhou** [umami-software/umami#2199] [Q&A]
I have been using the cloud version of Umami for some time and now I want to import the data into my self-hosted Umami service. 

How can I do that? 

I couldn't find the import feature on the page, like this one: https://cloud.umami.is/import

**Comment by funkhaus-wout** [comment]
Any updates on this? We are struggling with the same issue…
Also our database import does not work due to duplicate ID's and such.

**Reply by maique** [reply]
Would also love to get an answer to this one. Seems odd that importing is not possible.

**Reply by candidexmedia** [reply]
This is the closest thing I could find: https://github.com/umami-software/umami/issues/2456#issuecomment-2085404272

**Comment by mikecao** [comment]
You'll have to write a script to parse the CSV data and insert it into your database. There is no automatic tool for doing it.

**Comment by maique** [comment]
Thank you, that's what I was afraid of.

— maique

On January 29, 2025, gaurav ***@***.***> wrote:
> You'll have to write a script to parse the CSV data and insert it into
> your database. There is no automatic tool for doing it.
>

—
Reply to this email directly, view it on GitHub
<https://github.com/umami-
software/umami/discussions/2199#discussioncomment-11991063>, or
unsubscribe <https://github.com/notifications/unsubscribe-
auth/AAZ3TZS23WO624MWJHVJB3T2NBCSFAVCNFSM6AAAAABUZOOETKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCOJZGEYDMMY>.
You are receiving this because you commented.Message ID: <umami-
***@***.***>

**Comment by jrgong420** [comment]
Is there any guide or best practice how to import data from google analytics or plausible?

**Reply by candidexmedia** [reply]
This is definitely needed
